addons/portal/controllers/portal.py

```diff
@@ -196,6 +194,82 @@ def account(self, redirect=None, **post):
         response.headers['X-Frame-Options'] = 'DENY'
         return response
 
+    @http.route('/portal/attachment/add', type='json', auth='public')
```

Comment:
Now that I think about it, was there any reason to use a JSON route and serialize the data with `getDataURLFromFile`? 

We have had performance issues with this technique, as it is loads the whole file in browser memory. IIRC the browser used to freeze completely with files over 200-300MB, while it would work well with such files using a normal HTTP POST, like we do with `/web/binary/upload_attachment`.
You could have a look at the original fix odoo/enterprise@e000a2f3dd2626dd7b9c49967512d63c76d7361f

Maybe we could alter the new route with this in mind, and use an HTTP POST directly?

Reply:
I think the only reason is to standardize rpc to always be json, including not having to adapt an http route to return json.

But this is just a theorical reason, if it has an impact on what the user can upload then we can change it.

Reply:
Done

Reply:
Excellent, it works very well, I can now select and upload 800 MB files without having the browser freeze at all. And the error-reporting in case of broken upload is better too! :+1: